docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,382 @@
 
-As mentioned earlier, async code is a specification of work to be done in another context which needs to be explicitly started. Here are two primary ways to accomplish this:
+```fsharp
+let countFileBytes path =
+    async {
+        let! bytes = File.ReadAllBytesAsync(path) |> Async.AwaitTask
+        let fileName = Path.GetFileName(path)
+        printfn "File %s has %d bytes" fileName bytes.Length
+    }
+
+[<EntryPoint>]
+let main argv =
+    argv
+    |> Array.map countFileBytes
+    |> Async.Sequential
+    |> Async.RunSynchronously
+    |> ignore
+```
+
+This will execute `countFileBytes` for each item in `argv` in the order of the elements of `argv` rather than executing them in parallel.
 
-1. `Async.RunSynchronously` will start an async workflow on another thread and await its result.
+## Key async module functions
 
-    ```fsharp
-    open System
-    open System.Net
+When you write async code in F# you'll usually interact with a framework that handles scheduling of computations for you. However, this is not always the case, so it is good to learn the various starting functions to schedule asynchronous work.
 
-    let fetchHtmlAsync url =
-        async {
-            let uri = Uri(url)
-            use webClient = new WebClient()
-            let! html = webClient.AsyncDownloadString(uri)
-            return html
-        }
+Because F# asynchronous computations are a _specification_ of work rather than a representation of work that is already executing, they must be explicitly started with a starting function. There are many [Async starting functions](https://msdn.microsoft.com/library/ee370232.aspx) that are helpful in different contexts.The following section describes some of the more common starting functions.
 
-    // Execution will pause until fetchHtmlAsync finishes
-    let html = "https://dotnetfoundation.org" |> fetchHtmlAsync |> Async.RunSynchronously
+### Async.StartChild
 
-    // you actually have the result from fetchHtmlAsync now!
-    printfn "%s" html
-    ```
+Starts a child computation within an asynchronous computation. This allows multiple asynchronous computations to be executed simultaneously. The child computation shares a cancellation token with the parent computation. When the parent is canceled, the child is also canceled.
 
-2. `Async.Start` will start an async workflow on another thread, and will **not** await its result.
+Signature:
+
+```fsharp
+computation: Async<'T> * timeout: ?int -> Async<Async<'T>>
+```
 
-    ```fsharp
-    open System
-    open System.Net
+When to use:
 
-    let uploadDataAsync url data =
-        async {
-            let uri = Uri(url)
-            use webClient = new WebClient()
-            webClient.UploadStringAsync(uri, data)
-        }
+* When you want to execute multiple asynchronous computations concurrently rather than one at a time, but not necessarily in parallel.
```

Comment:
```suggestion
- When you want to execute multiple asynchronous computations concurrently rather than one at a time, but not necessarily in parallel.
```